src/abr/pensieve.cc

```diff
@@ -0,0 +1,105 @@
+#include "pensieve.hh"
+#include "ws_client.hh"
+
+using namespace std;
+
+//TODO: Reference already existing put_field()
+string put_field2(const uint16_t n)
+{
+  const uint16_t network_order = htobe16(n);
+  return string(reinterpret_cast<const char *>(&network_order),
+                sizeof(network_order));
+}
+
+//TODO: Reference already existing get_uint16()
+uint16_t get_uint16_2(const char * data)
+{
+  return be16toh(*reinterpret_cast<const uint16_t *>(data));
+}
+
+Pensieve::Pensieve(const WebSocketClient & client,
+         const string & abr_name, const YAML::Node & abr_config)
+  : ABRAlgo(client, abr_name)
+{
+
+    cout << abr_config << endl; // TODO: Actually use abr_config or remove it
+    // Config right now is done via constants in the Pensieve file, I am tempted
+    // to leave it that way unless we think we will actually use different
+    // configs for different people/channels.
```

Comment:
We probably won't change any constants, but you can add and read paths from abr_config to get rid of hardcoded paths.